### ISSUE TITLE
ComboBox Component

### ISSUE THREAD

**Issue by nikgraf** [nikgraf/belle#38]
# Basic example

```
<ComboBox>
  <Option value="apple">Apple</Option>
  <Option value="orange">Orange</Option>
<ComboBox>
```

# Behavior

- By default next to input there is a caret which will show all options, but the developer can disable it.
- By default the autocomplete shows an not entry found placeholder. This placeholder can be customized by providing a NoEntryPlaceholder component as child.

```
<Autocomplete>
  <NoEntryPlaceholder>No entry found.</NoEntryPlaceholder>
  <Option value="apple">Apple</Option>
  <Option value="orange">Orange</Option>
<Autocomplete>
```

- By default the autocomplete is enabled, but can be disabled with an disabled property. When disabled is active the input cannot be changed & styling is indicating that this component is disabled.
- By default the autocomplete is not case-sensitive & it finds matches everywhere in the string, but the developer can overwrite the matching function by providing the property: filter (typeOf func)
- By default all matched options are shown, but the matchedOptions can be limited via a parameter: maxOptions

- If the property `placeholder` (text) it will be used as placeholder for the input.
- If the property `defaultValue` (text) is defined the initial input value is set & can be changed.

# Behavior to be discussed

- Highlight the matched characters? Especially if we match parts of the string in the middle of it this can be really helpful to the user. Open question: how to do it with provided options? Inject a span?
Could be an optional feature. 
http://jsfiddle.net/adamboduch/jhZ6E/

- If we limit the amount of options we might want to display a hint about more options after the last one. Could be an optional feature.

- onChange function
Should the callback onChange fire once the user selected an option and return the value of the Option.
or
Should onChange fire once you type a new character into the input & we add a new callback onSelect for a selected option? 

# Internal structure

```
<div>

  <input type="text">
  
  optional: <span>carret</span> 
  
  <ul>
    <li>
      <Option />
    </li>
    …
    
    optional: no entry found placeholder
  </ul>

</div>
```

**Comment by jpuri** [comment]
Autocomplete has been renamed to combobox and is largely completed. But there are still a few points I was to note here so that we do not miss:
1. Right now we do not have a hover style for caret but will be probably a good thing to add.
2. Add option for highlighting matching text for combobox input in option - this looks good and we should have.
3. Right now when you navigate through options using keyboard their value is temporarily copied to combobox input. I had put it there coz many combobox out there I found have this feature. But it at times look misleading giving impression to user that value has been changed and is also error prone. We can think about removing it.
